app/routes/application.js

```diff
@@ -15,7 +16,19 @@ export default Ember.Route.extend(Analytics, OSFAgnosticAuthRouteMixin, {
     i18n: Ember.inject.service(),
     store: Ember.inject.service(),
     theme: Ember.inject.service(),
-
+    headTagsService: Ember.inject.service('head-tags'),
+    headTags: function() {
+        return [{
+            type: 'link',
+            attrs: {
+                rel: 'shortcut icon',
+                href: buildProviderAssetPath(config, this.get('theme.id'), 'favicon.ico', window.isProviderDomain)
+            }
+        }]
+    },
+    themeObserver: Ember.observer('theme.id', function() {
+        this.get('headTagsService').collectHeadTags();
+    }),
```

Comment:
It feels like all of this should go in the theme and here we should just do something like:
`headTags: this.get('theme.headTags')`

Thoughts?

Reply:
I like that, although i would be a bigger fan of it if we could also use the 'head-tags' handlebars helper within theme, but I don't know how we would go about doing that